submit.go

```diff
@@ -438,7 +438,11 @@ func saveLogPart(logNum int, filename string, reader io.Reader, reportDir string
 	// Either way, we need to append .gz, because we're compressing it.
 	var leafName string
 	if logRegexp.MatchString(filename) {
-		leafName = filename + ".gz"
+		if strings.HasSuffix(filename, ".gz") {
+			leafName = filename
+		} else {
+			leafName = filename + ".gz"
+		}
```

Comment:
personally I would find this more intuitive:

```suggestion
		leafName = filename               
		if !strings.HasSuffix(filename, ".gz") {
			leafName += "gz"
		}
```

... ymmv though.